source_gen/test/builder_test.dart

```diff
@@ -670,6 +670,41 @@ part of a;
 
 bar generated content
 
+foo generated content
+''',
+            ),
+          ),
+        },
+      );
+    });
+
+    test('includes preamble if enabled', () async {
+      await testBuilder(
+        const CombiningBuilder(
+          preamble: '''
+foo
+
+bar''',
```

Comment:
[nit] Adding a comment in the preamble would be more realistic.
```suggestion
// foo

// bar''',
```